misk/src/main/kotlin/misk/jdbc/DataSourceModule.kt

```diff
@@ -0,0 +1,60 @@
+package misk.jdbc
+
+import com.google.inject.Key
+import com.zaxxer.hikari.HikariConfig
+import com.zaxxer.hikari.HikariDataSource
+import misk.inject.KAbstractModule
+import misk.inject.asSingleton
+import javax.inject.Inject
+import javax.inject.Provider
+import javax.sql.DataSource
+import kotlin.reflect.KClass
+
+/** Binds a [DataSource] configured from the specific entry in the [DataSourcesConfig] */
+class DataSourceModule constructor(
+  private val datasourceName: String,
+  private val key: Key<DataSource>
+) : KAbstractModule() {
+
+  override fun configure() {
+    bind(key).toProvider(DataSourceProvider(datasourceName)).asEagerSingleton()
+  }
+
+  companion object {
+    fun create(datasourceName: String) =
+        DataSourceModule(datasourceName, Key.get(DataSource::class.java))
+
+    fun create(datasourceName: String, annotatedBy: Annotation) =
+        DataSourceModule(datasourceName, Key.get(DataSource::class.java, annotatedBy))
+
+    fun <A : Annotation> create(datasourceName: String, annotatedBy: Class<A>) =
+        DataSourceModule(datasourceName, Key.get(DataSource::class.java, annotatedBy))
+
+    fun <A : Annotation> create(datasourceName: String, annotatedBy: KClass<A>) =
+        DataSourceModule(datasourceName, Key.get(DataSource::class.java, annotatedBy.java))
+  }
+
+  private class DataSourceProvider(private val datasourceName: String) : Provider<DataSource> {
```

Comment:
should this be in a separate hikari package? do you think apps would want to provide their own Datasource?

Reply:
If so, they can implement their own module for it. The only thing this module does is setup the Hikari CP, so not much value in abstracting that out